src/main/java/io/tus/java/client/TusClient.java

```diff
@@ -286,4 +318,16 @@ public void prepareConnection(@NotNull HttpURLConnection connection) {
             }
         }
     }
+    
+    /**
+     * Actions to be performed after a successful upload completion. 
+     * Manages URL removal from the URL store if remove fingerprint on success is enabled
+     * 
+     * @param upload that has been finished
+     */
+    public void uploadFinished(@NotNull TusUpload upload) {
```

Comment:
I suggest to use protected as the method visibility. So, the TusClient extensions can also modify this behaviour.

Reply:
:+1: